sdk/batch/Azure.Compute.Batch/samples/Sample1_CreatePool_Job_Task.md

```diff
@@ -0,0 +1,96 @@
+```
+
+With the BatchAccountResource you can create a pool with the [batchAccount.GetBatchAccountPools().CreateOrUpdateAsync](https://learn.microsoft.com/dotnet/api/azure.resourcemanager.batch.batchaccountpoolcollection.createorupdateasync?view=azure-dotnet) command
+```C# Snippet:Batch_Sample01_PoolCreation
+var poolName = "HelloWorldPool";
+var imageReference = new BatchImageReference()
+{
+    Publisher = "canonical",
+    Offer = "0001-com-ubuntu-server-jammy",
+    Sku = "22_04-lts",
+    Version = "latest"
+};
+string nodeAgentSku = "batch.node.ubuntu 22.04";
+
+BatchAccountPoolResource pool = (await batchAccount.GetBatchAccountPools().CreateOrUpdateAsync(WaitUntil.Completed, poolName, new BatchAccountPoolData()
+{
+    VmSize = "Standard_DS1_v2",
+    DeploymentConfiguration = new BatchDeploymentConfiguration()
+    {
+        VmConfiguration = new BatchVmConfiguration(imageReference, nodeAgentSku)
+    },
+    ScaleSettings = new BatchAccountPoolScaleSettings()
+    {
+        FixedScale = new BatchAccountFixedScaleSettings()
+        {
+            TargetDedicatedNodes = 1
+        }
+    }
+})).Value;
+```
+
+## Authenticating the Azure.Compute.Batch `BatchClient`
+
+Creation of Batch jobs and tasks can only be preformed with the `Azure.Compute.Batch` sdk.  A `BatchClient` object is needed to preform Batch operations and can be created using [Microsoft Entra ID authtentication](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/identity/Azure.Identity/README.md) and the Batch account endpoint  
+
+```C# Snippet:Batch_Sample01_CreateBatchClient
+var credential = new DefaultAzureCredential();
+BatchClient _batchClient = new BatchClient(
+new Uri("https://examplebatchaccount.eastus.batch.azure.com"), credential);
+```
+
+### Job creation
+
+Before we can create Batch Tasks we first need to create a Job for the tasks to be associatd with, this can be done via the `CreateJobAsync` command. The basic elements needed are a Id for job itself and the name of the pool that this job will run against. 
+```C# Snippet:Batch_Sample01_CreateBatchJob
+await _batchClient.CreateJobAsync(new BatchJobCreateContent("jobId", new BatchPoolInfo() { PoolId = "poolName" }));
+```
+
+### Task creation
+
+Batch tasks can be created from the BatchClient via the `CreateTaskAsync`.  The basic elements needed are the name of the job the task will be assigned to, and id for the task itself, and a command to run.
+```C# Snippet:Batch_Sample01_CreateBatchTask
+await _batchClient.CreateTaskAsync("jobId", new BatchTaskCreateContent("taskId", $"echo Hello world"));
+```
+### Task results
+
+Onces the tasks are complete `GetTasksAsync` cand be used to retrieve the `BatchTask` object and `GetTaskFileAsync` can be used to get the output files
+
+```C# Snippet:Batch_Sample01_GetTasks
+var completedTasks = _batchClient.GetTasksAsync(jobId, filter: "state eq 'completed'");
```

Comment:
What is `_batchClient` and how do I make one?   Snippets are intended to be fully copy/paste and should be entirely self-contained and complete.   

Developers shouldn't have to go elsewhere and find other boilerplate to make a snippet work - just paste into their app and compile.

Reply:
I was kind of following the keyvalut examples where they state that all the samples will use the client created above https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/keyvault/Azure.Security.KeyVault.Secrets#examples